components/library/DateRangePicker/Preview/DateRangePicker.jsx

```diff
@@ -0,0 +1,226 @@
+import React, { useState, useCallback, useEffect } from 'react';
+import {
+  Popover,
+  Button,
+  TextField,
+  Box,
+  DatePicker,
+  Icon,
+  OptionList,
+  Scrollable,
+  InlineGrid,
+  BlockStack,
+  InlineStack
+} from '@shopify/polaris';
+import { ArrowRightIcon, CalendarIcon } from '@shopify/polaris-icons';
+
+export const DateRangePicker = ({ onDateRangeSelect }) => {
+  const [windowWidth, setWindowWidth] = useState(window.innerWidth);
+  const [popoverActive, setPopoverActive] = useState(false);
+  const today = new Date(new Date().setHours(0, 0, 0, 0));
+  const yesterday = new Date(
+    new Date(new Date().setDate(today.getDate() - 1)).setHours(0, 0, 0, 0)
+  );
+  const ranges = [
+    { title: 'Today', period: { since: today, until: today } },
+    { title: 'Yesterday', period: { since: yesterday, until: yesterday } },
+    {
+      title: 'Last 7 days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 7)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Last 30 days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 30)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Last 90 days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 90)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Last 365 Days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 365)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Custom',
+      period: { since: yesterday, until: yesterday }
+    }
+  ];
+  const [activeDateRange, setActiveDateRange] = useState(ranges[0]);
+  const [dateState, setDateState] = useState({
+    month: activeDateRange.period.since.getMonth(),
+    year: activeDateRange.period.since.getFullYear()
+  });
+
+  const handleMonthChange = useCallback((month, year) => {
+    setDateState({ month, year });
+  }, []);
+
+  const formatDate = (date) => date.toISOString().split('T')[0];
+
+  useEffect(() => {
+    const handleResize = () => {
+      setWindowWidth(window.innerWidth);
+    };
+    window.addEventListener('resize', handleResize);
+    return () => window.removeEventListener('resize', handleResize);
+  }, []);
+
+  useEffect(() => {
+    setDateState({
+      month: activeDateRange.period.since.getMonth(),
+      year: activeDateRange.period.since.getFullYear()
+    });
+  }, [activeDateRange]);
+
+  const isMobile = windowWidth < 768;
```

Comment:
So there is actually a hook you can import from Polaris called `useBreakpoints` that gives you the breakpoints you need for determining screen size. So you should be able to remove everything related to `windowWidth`/`isMobile` and do this:

`const { mdDown } = useBreakpoints()`

Then wherever you use `isMobile`, just use the `mdDown` var instead.

You can look at all the other breakpoint values available by just logging out the value coming from `useBreakpoints`, but it looks like `mdDown` is the exact breakpoint (767px and down) you're replicating here with the `windowWidth` code.

Reply:
Yeah , useBreakpoints was way easier to use! Just included that